doc/Programming-guide/Key-platform-features/Common-Type-Library/index.md

```diff
@@ -33,38 +67,105 @@ Matching FQN in a different Tenant | N/A | OK | OK
+
+To create a new tenant CT, click **Add new type** and then fill in all the required fields. Click **Add** to apply the changes.
+
+![Create Tenant CTL](attach/create_tenant_ctl.png)
+
+To view a CT, open the CT details window by clicking the row in the list in the **Tenant** CTL window.
+
+![View Tenant CTL](attach/view_tenant_ctl.png)
+
+To view another version of the CT, click on the version drop-down and select the appropriate version. To create a new version of the tenant CT of the same FQN, click **Create new version** at the top of the CT details window. To export the CT, click **Export** and select an export method from the drop-down. See [CT schema import and export](#ct-schema-import-and-export) for export details. To delete the tenant CT version, click Delete at the top of the CT details window.
+
+</div>
+
+<div id="managing-application-ctl" class="tab-pane fade" markdown="1" >
+
+### Managing Application CTL
+
+The **Application CTL** window displays a list of CTs of the current application, of the current tenant and all CTs with system scope. The tenant developer can add new application CTs and delete application CTs from the system. It has read-only access to the system CTs and to the tenant CTs within **Application CTL** window. Tenant developer is still able to manage tenant CTs within **Tenant CTL** window.
+
+![Application CTL](attach/application_ctl.png)
+
+To display/hide higher scopes of CTs, check/uncheck **Display higher scopes** checkbox.
+
+To create a new application CT, click **Add new type** and then fill in all the required fields. Click **Add** to apply the changes.
+
+![Create Application CTL](attach/create_application_ctl.png)
+
+To view a CT, open the CT details window by clicking the row in the list in the **Application CTL** window.
+
+![View Application CTL](attach/view_application_ctl.png)
+
+To view another version of the CT, click on the version drop-down and select the appropriate version. To create a new version of the application CT of the same FQN, click **Create new version** on the top of CT details. To export the CT, click **Export** and select export method from drop-down. See [CT schema import and export](#ct-schema-import-and-export) for export details. To delete the application CT version, click **Delete** on the top of the CT details. To promote the application CT from application to tenant scope, click **Promote**.
+
+</div>
+</div>
+
+## CT promotion ##
+
+Users are able to promote the CT's scope from APPLICATION to TENANT, provided that there is no other CT with the identical FQN within the given tenant, and that they have a permission to do so. Otherwise an appropriate error will be displayed. Promotion procedure updates scope of CT including all its versions. Promotion of CT can be performed in two ways: via [REST API]({{root_url}}Programming-guide/Server-REST-APIs#) call or by clicking **Promote** in the **Application CTL** window of Administration UI as shown below.
```

Comment:
"Otherwise an appropriate error will be displayed." - are we only talking about UI here?